### DISCUSSION TITLE
Getting formatted text to appear in published content

### DISCUSSION THREAD

**Discussion by domcorleone21** [danielsaidi/RichTextKit#185] [Q&A]
Hi, I'm building a writing app and I have set up the RichTextKit in my app but it the formatted text only appears on the new story/edit page when I am creating new text but once I publish the formatted text doesn't appear formatted in my published story. is there something I have to pass through the .init for it to work properly?

Apologies in advance. I'm still new to developing so I don't know how to properly explain things just yet. Any help would be appreciated!

**Comment by danielsaidi** [comment]
Hi @domcorleone21 

Hmmmm, I'm not sure what can be causing this. Have you checked out the demo app for some clues?

**Reply by domcorleone21** [reply]
Hi @danielsaidi 

Thanks for getting back to me. I did check out the demo app but as I mentioned I am still new to developing so I wasn't sure what to look for. Should I be using HTML or NSAttributedString when looking to decode/encode the rich text? I am using Firebase and have a richText field that receives the data as a NSAttributedString but isnt not decoding it properly. 

Any part of the demo code you would recommend I look into specifically? Any help would be GREATLY appreciated

**Reply by danielsaidi** [reply]
I have designed this with `NSAttributedString` in min, although I think that type has ways to convert the string to and from HTML.

I haven't had the possibility to work with this for a few months, so my memory is a bit bad regarding these parts. I'm also on vacation for a few weeks, so it'd be great if you could take a look in the docs for more info.

There are things in the SDK to export a string to various formats, so if you look for "rich text format" or perhaps check out the RichTextAction for an export action, that might help you get started.
